pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -215,6 +215,10 @@ func getDefaultCNINetwork(confDir string, binDirs []string) (*cniNetwork, error)
 	return nil, fmt.Errorf("no valid networks found in %s", confDir)
 }
 
+func noSuchNetNS(err error) bool {
+	return strings.Contains(err.Error(), "no such file or directory") && strings.Contains(err.Error(), "failed to Statfs")
```

Comment:
If possible I would prefer checking the error code for the "well-known" error codes see: https://github.com/containernetworking/cni/blob/master/pkg/types/types.go#L139 + https://github.com/containernetworking/cni/blob/master/SPEC.md#well-known-error-codes instead of parsing the strings (which will probably not that performant).

Also I think this should be moved to a common place rather than be duplicated in code e.g. somewhere here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/network/plugins.go

Reply:
Hm it seems like these codes are not available in this method :(

Reply:
I don't think the message checking code here is right. Either `no such file or directory` or `failed to Statfs` should be considered as network disabled or missing.

Reply:
And I'd prefer to make `noSuchNetNS` as a common util function.

Reply:
Both the `no such file or directory` and `failed to Statfs` cannot ensure the NetNS does not exist if the cni plugin does not invoke the oommonly-used function `ns.GetNS()`